packages/block-editor/src/store/test/reducer.js

```diff
@@ -3085,18 +3080,6 @@ describe( 'state', () => {
 				'afd1cb17-2c08-4e7a-91be-007ba7ddc3a1': { food: 'banana' },
 			} );
 		} );
-
-		it( 'returns null on anything other than block attributes update', () => {
```

Comment:
Returning "null" or "undefined" from a reducer is wrong because it creates a lot of bugs (we still have one instance like that in our code). I can't find the exact reason why this was used here.

For instance here, if a random action is triggered like "SHOW_INSERTION_POINT" because the user moved the mouse slightly while typing, it would cause an "undo" state because that particular reducer resetted its value to null because it doesn't know about this random action "SHOW_INSERTION_POINT". So you need to either exclude all the actions that shouldn't be impact the reducer or always return state and only return "null" for the explicit list of actions that you want to reset the state with. It's always better to do the latter because developers can randomly add actions to a store and they don't expect these new actions to have an impact on a reducer they didn't touch.

Reply:
cc @ntsekouras cause he asked a similar question in another PR.